core/src/main/java/hudson/Functions.java

```diff
@@ -665,7 +627,24 @@ public static boolean isAutoRefresh(HttpServletRequest request) {
     public static boolean isCollapsed(String paneId) {
     	return PaneStatusProperties.forCurrentUser().isCollapsed(paneId);
     }
-    
+
+    public static boolean useUserTimeZone() {
+        return TimeZoneProperty.forCurrentUser() != null;
+    }
+
+    public static String getUserTimeZone() {
+        return TimeZoneProperty.forCurrentUser();
+    }
+
+    public static String getUserTimeZonePostfix() {
+        if (!useUserTimeZone()) {
+            return "";
+        }
+
+        TimeZone tz = TimeZone.getTimeZone(getUserTimeZone());
```

Comment:
What will happen here if the string cannot be converted? Looks like some error handling and protective code is missing

Reply:
Per Java docs, it'll just use GMT.

Reply:
(Also just tested to confirm)

Reply:
Shouldn't it use the system one, which is the one this is overriding?

Reply:
@varyvol actually that could be debated. In terms of Jenkins, using the system one if I have an invalid user preference vs using GMT makes zero difference to me personally. The company I'm at now has offices in PST, MST, EET, ICT. It doesn't make my life any easier if it's one of those I'm not in vs GMT. I think I could do it, but not sure I understand the benefit for introducing this extra bit of complexity. 

@alecharp I'll see about adding a unit test. I have to figure out the Jenkins testing framework.

Reply:
@silent-snowman maybe the benefits are not great (I guess it's debatable depending on specific situations) but I think changing the current behaviour would actually cause strangeness at least.

Reply:
@varyvol I think this change is terrible, but whatever it's not my project. I hate that you are making me put my name next to a shitty change because it seems like you won't approve the PR without it. Not to mention it's quite nitpicky.